app/services/subscriptions/validate_service.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Subscriptions
+  class ValidateService < BaseValidator
+    def valid?
+      valid_customer?
+      valid_plan?
+      valid_subscription_date?
+
+      if errors?
+        result.validation_failure!(errors: errors)
+        return false
+      end
+
+      true
+    end
+
+    private
+
+    def valid_customer?
+      return true if args[:customer]
+
+      add_error(field: :customer, error_code: 'customer_not_found')
```

Comment:
That's were our current validator implementation is limited (cc @jdenquin :trollface:)

For me if we are not able to find the customer or the plan, we should render a 404 and not a 422.

Maybe we should handle this at the `BaseValidator` level. Let me know if you need help on this topic.

Reply:
I am never sure what to return in such a cases :) 404 makes sense but on the other hand it's 'just' relationship, not the main object as in e.g. GET show 😅 However, refactored it a bit and returning 404 now